### DISCUSSION TITLE
Signer Liveness Enforcement in Nakamoto

### DISCUSSION THREAD

**Discussion by AshtonStephens** [stacks-network/stacks-core#4195] [Blockchain]
# Signer Liveness Enforcement in Nakamoto

![Priority High](https://img.shields.io/badge/priority-high-red) ![Required For SIP](https://img.shields.io/badge/Required_for_SIP-yes-red)

We need to determine how to best enforce signer liveness in the Nakamoto SIP.

- Proposed way to deal with signer liveness in sBTC: [Draft Markdown link](https://github.com/stacksgov/sips/blob/feat/sip-025-sbtc/sips/sip-025/sip-025-sbtc.md#proposed-solution)
 
We **may** be able to reuse all of it; this discussion is here to get us all aligned.

> Details to follow…

**Comment by AshtonStephens** [comment]
Within the [sBTC SIP](https://github.com/stacksgov/sips/blob/feat/sip-025-sbtc/sips/sip-025/sip-025-sbtc.md#stacker-authorization-transaction) witholding PoX payout to stackers who are not live serves as the enforcement that stackers remain online. The sBTC SIP refers specifically to the signing of the the deposit / withdrawal requests but the signing bitmap feature could be used for requiring that stackers sign stacks blocks as well.

> The signing bitmap will include a map of both those who vote to accept and those who vote to reject because the set of Stackers must prove that at least 70% of them by weight were online when the vote was taken; if less than 70% of signers by weight are voting then this is a liveness failure on the Stackers' part, and both their PoX payouts and peg-ins will be stopped until Stackers can reach quorum.

Is this solution sufficient for maintaining signer liveness for Nakamoto?

**Comment by AshtonStephens** [comment]
@jferrant @kantai @jcnelson this is the best place to continue discussing liveness enforcement. It relates to https://github.com/stacks-network/stacks-core/issues/4248

**Comment by philiphacks** [comment]
From the Stacking DAO perspective, we'd love to be included in the discussions on this

Liveness and penalties are important for building a robust signer network and from the Stacking DAO point of view we'd love to quantify the performance of each signer which helps us with delegating STX stake proportionate to the performance of a signer.

**Comment by towny694** [comment]
Writers of the West is a prestigious firm with over a decade of experience in book **[ghostwriting services](https://writersofthewest.net/)** and publishing across the USA, Canada, and Europe. With offices in Houston, New York, and Los Angeles, they have written over 170 bestsellers in over 10 genres in the past ten years. Their vision is to provide affordable ghost writing services that make a difference. Over the last decade, they have helped more than 2500 aspiring authors tell their stories.
